src/Http/Http.Abstractions/src/HttpResponse.cs

```diff
@@ -93,6 +98,12 @@ public abstract class HttpResponse
         /// <param name="disposable">The object to be disposed.</param>
         public virtual void RegisterForDispose(IDisposable disposable) => OnCompleted(_disposeDelegate, disposable);
 
+        /// <summary>
+        /// Registers an object for disposal by the host once the request has finished processing.
+        /// </summary>
+        /// <param name="disposable">The object to be disposed.</param>
+        public virtual void RegisterForDispose(IAsyncDisposable disposable) => OnCompleted(_disposeAsyncDelegate, disposable);
```

Comment:
If the type implements both then the compiler will consider this ambiguous and error. RegisterForDisposeAsync?

Reply:
Agreed. We should rename this to `RegisterForDisposeAsync` to avoid that ambiguity.